### PR TITLE
Show only GH Enterprise notifications on GH Enterprise

### LINKED ISSUE #2065: Unread notifications from GHE appear on Github and vice versa
Can we display notifications marked as unread from the same domain only?

### PR DESCRIPTION
**Pull request by simonkotwicz**
Closes #2065

### CODE REVIEW

**Comment by simonkotwicz** [comment]
Actually, I might need to modify this PR, I forgot about gist.github.com.

**Comment by bfred-it** [comment]
>  I forgot about `gist.github.com`

That's fine, there are no notifications on that domain

**Comment by simonkotwicz** [comment]
There is a notification icon though, do we not care about seeing unread notifications there?

**Comment by bfred-it** [comment]
Good point. You can use the `URL` constructor to parse the `url` variable and match the main domain only, e.g. `github.com` instead of the whole origin.

**Comment by bfred-it** [comment]
Actually that's a bad idea because `url` could be `hello.co.uk` and currentDomain` might be `wow.co.uk`, which we have to turn into `co.uk` to make my logic work.

This needs to explicitly account for `gist.github.com` instead

File: source/features/mark-unread.tsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ async function getNotifications(): Promise<Notification[]> {
 	const {unreadNotifications} = await browser.storage.local.get({
 		unreadNotifications: []
 	});
-	return unreadNotifications;
+	return unreadNotifications.filter(({url}: AnyObject) => location.hostname.endsWith(new URL(url).hostname));
```
Comment (bfred-it) [inline source/features/mark-unread.tsx]: That's a good solution

```suggestion
	return unreadNotifications.filter(({url}: Notification) => location.hostname.endsWith(new URL(url).hostname));
```

**Comment by bfred-it** [comment]
This doesn't break github.com and gist.github.com, so it's a-ok for me

**Comment by busches** [comment]
I never have notifications on GHE. I'm not sure if they're not fully baked yet or our instance isn't configured to use them. Either way, this looks fine to me.

**Pull request merged**